### PR TITLE
Clean Up: Variable Abstraction, Scoping, and Simplification

### PR DESCRIPTION
**Pull request by talbs**
This work does a few things around Sass variables within the Pattern Library:

* syncs up variable naming standards with several already set variables (buttons and forms primarily)
* plumbs through variables to be used consistently in future theming where they weren't set (base resets and buttons)
* abstracts out a few variables that were being used in unconventional or out of scope ways
* simplifies button-based rules for easier maintenance and for Sass extension

- - -

##### Reviewers

* [x] FED/UI Architecting - @frrrances

### CODE REVIEW

**Comment by talbs** [comment]
@frrrances, this has been rebased off of master to catch all of the big icon changes, FYI.

File: _posts/patterns/2015-04-09-buttons.md [hunk 1]
```diff
@@ -19,78 +19,85 @@ info: Buttons should be used for performing actions within the edX environment.
...
+        <button type="button" class="btn-overlay btn-base">Overlay Base</button>
+        <button type="button" class="btn-overlay btn-large">Overlay Large</button>
+        <button type="button" class="btn-overlay btn-small">Overlay Small</button>
+        <button type="button" class="btn-overlay btn-base" disabled>Overlay Disabled</button>
     </div>
 </div>
 
 <h3 class="hd-6 example-set-hd">Links</h3>
 <div class="example-set">
+<<<<<<< HEAD
```
Comment (frrrances) [inline _posts/patterns/2015-04-09-buttons.md]: Oops! You still gots the diff in here...

File: pattern-library/sass/patterns/_buttons.scss [hunk 2]
```diff
@@ -239,7 +241,8 @@
 // #ELEVATED-ALT
 // ----------------------------
 .btn-elevated-alt {
-    box-shadow: inset 0 0 -$buttons-elevated-shadow 0 0 palette(grayscale-cool, light);
+    @extend %btn;
+    box-shadow: inset 0 0 -$btn-elevated-shadow 0 0 palette(grayscale-cool, light);
```
Comment (frrrances) [inline pattern-library/sass/patterns/_buttons.scss]: ooo custom button shadow size. fancee.

File: pattern-library/sass/utilities/_variables.scss [hunk 3]
```diff
@@ -359,98 +359,202 @@ $input-width: (
...
+$btn-border-size:                           1px !default;
+$btn-shadow:                                3px !default;
+$btn-font-weight:                           font-weight(semi-bold) !default;
+$btn-border-radius:                         $component-border-radius !default;
+
+// default button
+$btn-default-border-color:                  palette(primary, base) !default;
+$btn-default-background:                    palette(grayscale, white) !default;
+$btn-default-color:                         palette(primary, base) !default;
+// $btn-default-shadow:                     ;
```
Comment (frrrances) [inline pattern-library/sass/utilities/_variables.scss]: Why the blank rules?

**Comment by frrrances** [comment]
Things are looking very themeable in here. I like it. I left a couple small nits/notes, but overall this looks ready to :+1:

File: _posts/patterns/2015-04-09-buttons.md (hunk 1 above)
Comment (talbs) [inline _posts/patterns/2015-04-09-buttons.md]: giphy

**D'oh!** Thanks for the catch!

File: pattern-library/sass/utilities/_variables.scss (hunk 3 above)
Comment (talbs) [inline pattern-library/sass/utilities/_variables.scss]: Oh, I added these based on the button properties you and I defined within the theming inventory (https://docs.google.com/a/edx.org/spreadsheets/d/1kHttF70rf2p7-9-Ev6kubfSFwncrDi0KzwFO9jArOx0/edit?usp=sharing). 

My hope was that it was easier to sync up when we got to the platform, but if its just noise/junk, I can lose it until or if we use it.

**Comment by talbs** [comment]
@frrrances, merged this in, but I left you some reasonings/background to your comments. Happy to follow-up to remove those ghost variables - just shout!

File: pattern-library/sass/utilities/_variables.scss (hunk 3 above)
Comment (frrrances) [inline pattern-library/sass/utilities/_variables.scss]: Sounds good. Thanks for the 411.

**Pull request merged**